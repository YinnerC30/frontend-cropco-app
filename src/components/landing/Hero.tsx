import { Link } from 'react-router-dom';

export default function Hero() {
  return (
    <section className="py-20 bg-green-50 dark:bg-green-200">
      <div className="container px-4 mx-auto text-center">
        <h1 className="mb-6 text-4xl font-bold text-green-800 md:text-5xl">
          Administra tus cultivos con CropCo
        </h1>
        <p className="mb-8 text-xl text-gray-600">
          La solución integral para gestionar la información de tus cultivos
          agrícolas de manera eficiente y efectiva.
        </p>
        <Link
          to={'/app/authentication/login'}
          className="px-8 py-3 text-lg text-white bg-green-600 rounded-lg hover:bg-green-700"
        >
          Comienza ahora
        </Link>
      </div>
    </section>
  );
}
