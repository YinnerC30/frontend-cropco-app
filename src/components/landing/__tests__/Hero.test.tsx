import { render, screen } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import Hero from '../Hero';

describe('Hero Component', () => {
  it('should render correctly', () => {
    const { container } = render(<Hero />);

    // Verificar que el componente se renderice
    const heroSection = container.querySelector('section');
    expect(heroSection).toBeInTheDocument();
  });

  it('should display the main heading', () => {
    render(<Hero />);

    const mainHeading = screen.getByRole('heading', {
      name: /administra tus cultivos con cropco/i,
    });
    expect(mainHeading).toBeInTheDocument();
    expect(mainHeading).toHaveClass(
      'mb-6',
      'text-4xl',
      'font-bold',
      'md:text-5xl',
      'text-slate-900'
    );
  });

  it('should display the description text', () => {
    render(<Hero />);

    const description = screen.getByText(
      /la solución integral para gestionar la información de tus cultivos agrícolas de manera eficiente y efectiva/i
    );
    expect(description).toBeInTheDocument();
    expect(description).toHaveClass('mb-8', 'text-xl', 'text-gray-600');
  });

  it('should have the correct section structure and classes', () => {
    const { container } = render(<Hero />);

    const section = container.querySelector('section');
    expect(section).toHaveClass('py-20', 'bg-primary/60', 'dark:bg-black-200');

    const containerDiv = section?.querySelector('.container');
    expect(containerDiv).toBeInTheDocument();
    expect(containerDiv).toHaveClass('px-4', 'mx-auto', 'text-center');
  });

  it('should have proper heading hierarchy', () => {
    render(<Hero />);

    const heading = screen.getByRole('heading', { level: 1 });
    expect(heading).toBeInTheDocument();
    expect(heading.tagName).toBe('H1');
  });

  it('should match snapshot', () => {
    const { container } = render(<Hero />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
