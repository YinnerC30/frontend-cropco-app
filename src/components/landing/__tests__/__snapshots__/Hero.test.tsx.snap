// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Hero Component > should match snapshot 1`] = `
<section
  class="py-20 bg-primary/60 dark:bg-black-200"
>
  <div
    class="container px-4 mx-auto text-center"
  >
    <h1
      class="mb-6 text-4xl font-bold md:text-5xl text-slate-900"
    >
      Administra tus cultivos con CropCo
    </h1>
    <p
      class="mb-8 text-xl text-gray-600"
    >
      La solución integral para gestionar la información de tus cultivos agrícolas de manera eficiente y efectiva.
    </p>
  </div>
</section>
`;
